### DISCUSSION TITLE
Missing attribute - Vector3(pip install) & make check faliure in python(source build)

### DISCUSSION THREAD

**Discussion by KPK101** [NanoComp/meep#1980] [Q&A]
Hi, I have installed meep through python and when I try to run a file containing Vector3 usage, I get this error
<img width="738" alt="Screenshot 2022-03-03 at 6 25 27 PM" src="https://user-images.githubusercontent.com/70443224/156569180-d8592a82-7bc8-4376-ad79-49f39df78f1e.png">

**Comment by KPK101** [comment]
I tried to build from the source, but when running the make check for meep, this error popped up. 

============================================================================
Testsuite summary for meep 1.22.0
============================================================================
# TOTAL: 23
# PASS:  23
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
Making check in scheme
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-am
make[2]: Nothing to be done for `check-am'.
Making check in libpympb
make[1]: Nothing to be done for `check'.
Making check in python
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-am
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-TESTS
FAIL: tests/test_3rd_harm_1d.py
FAIL: tests/test_absorber_1d.py
FAIL: tests/test_adjoint_solver.py
FAIL: tests/test_adjoint_cyl.py
FAIL: tests/test_adjoint_jax.py
FAIL: tests/test_antenna_radiation.py
FAIL: tests/test_array_metadata.py
FAIL: tests/test_bend_flux.py
FAIL: tests/test_binary_partition_utils.py
FAIL: tests/test_binary_grating.py
FAIL: tests/test_cavity_arrayslice.py
FAIL: tests/test_cavity_farfield.py
FAIL: tests/test_chunk_balancer.py
FAIL: tests/test_chunk_layout.py
FAIL: tests/test_chunks.py
FAIL: tests/test_conductivity.py
FAIL: tests/test_cyl_ellipsoid.py
FAIL: tests/test_dft_energy.py
FAIL: tests/test_dft_fields.py
FAIL: tests/test_diffracted_planewave.py
FAIL: tests/test_dispersive_eigenmode.py
FAIL: tests/test_divide_mpi_processes.py
FAIL: tests/test_dump_load.py
FAIL: tests/test_eigfreq.py
FAIL: tests/test_faraday_rotation.py
FAIL: tests/test_field_functions.py
FAIL: tests/test_force.py
FAIL: tests/test_fragment_stats.py
FAIL: tests/test_gaussianbeam.py
FAIL: tests/test_geom.py
FAIL: tests/test_get_point.py
FAIL: tests/test_get_epsilon_grid.py
FAIL: tests/test_holey_wvg_bands.py
FAIL: tests/test_holey_wvg_cavity.py
FAIL: tests/test_kdom.py
FAIL: tests/test_ldos.py
FAIL: tests/test_material_dispersion.py
FAIL: tests/test_material_grid.py
FAIL: tests/test_medium_evaluations.py
FAIL: tests/test_mpb.py
FAIL: tests/test_mode_coeffs.py
FAIL: tests/test_mode_decomposition.py
FAIL: tests/test_multilevel_atom.py
FAIL: tests/test_n2f_periodic.py
FAIL: tests/test_oblique_source.py
FAIL: tests/test_physical.py
FAIL: tests/test_prism.py
FAIL: tests/test_pw_source.py
FAIL: tests/test_refl_angular.py
FAIL: tests/test_ring.py
FAIL: tests/test_ring_cyl.py
FAIL: tests/test_simulation.py
FAIL: tests/test_special_kz.py
FAIL: tests/test_source.py
FAIL: tests/test_timing_measurements.py
FAIL: tests/test_user_defined_material.py
FAIL: tests/test_visualization.py
FAIL: tests/test_wvg_src.py
============================================================================
Testsuite summary for meep 1.22.0
============================================================================
# TOTAL: 58
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  58
# XPASS: 0
# ERROR: 0
============================================================================
See python/test-suite.log
============================================================================
make[4]: *** [test-suite.log] Error 1
make[3]: *** [check-TESTS] Error 2
make[2]: *** [check-am] Error 2
make[1]: *** [check] Error 2
make: *** [check-recursive] Error 1

**Comment by smartalecH** [comment]
Is your `PYTHONPATH` set?

https://meep.readthedocs.io/en/latest/Installation/#building-from-source

**Comment by KPK101** [comment]
I installed everthing in the global default and did not set python path. Also, why is the simple `pip install meep` not working correctly (missing attributes)?

**Reply by smartalecH** [reply]
`pip install meep` won't work because it's not supported. You are probably installing some other package whose name is also `meep`. 

You can use `conda`, as described in the docs [here](https://meep.readthedocs.io/en/latest/Installation/#conda-packages). 

If you built from source, you need to set your PYTHONPATH, as indicated by your `make check` log.

The docs cover these issues pretty thoroughly.
